var/spack/repos/builtin/packages/occa/package.py

```diff
@@ -33,91 +38,177 @@ class Occa(Package):
     version("0.2.0", tag="v0.2.0", commit="2eceaa5706ad6cf3a1b153c1f2a8a2fffa2d5945")
     version("0.1.0", tag="v0.1.0", commit="381e886886dc87823769c5f20d0ecb29dd117afa")
 
+    depends_on("hip", when="+rocm")
     depends_on("c", type="build")  # generated
     depends_on("cxx", type="build")  # generated
     depends_on("fortran", type="build")  # generated
+    depends_on("cmake@3.30:", type="build", when="build_system=cmake")
+    depends_on("gmake", type="build")
 
-    variant("cuda", default=True, description="Activates support for CUDA")
-    variant("openmp", default=True, description="Activates support for OpenMP")
-    variant("opencl", default=True, description="Activates support for OpenCL")
+    build_system(
+        conditional("cmake", when="@1.1.0:"),
+        conditional("makefile", when="@:1.0.9"),
+        default="makefile",
+    )
 
-    depends_on("cuda", when="+cuda")
-    depends_on("gmake", type="build")
+    variant("openmp", default=False, description="Enable support for OpenMP")
+    variant("debug", default=False, description="Enable a build with debug symbols")
 
+    # HIP build fails with Make on some systems. So, we disable it when
+    # Make based build is used.
+    conflicts("+rocm", when="@:1.0.9")
+    # OpenMP build fails with Make on some systems. So, we disable it when
+    # Make based build is used.
+    conflicts("+openmp", when="@:1.0.9")
+    # gcc and cuda conflicts.
     conflicts("%gcc@6:", when="^cuda@:8")
     conflicts("%gcc@7:", when="^cuda@:9")
 
-    def install(self, spec, prefix):
-        # The build environment is set by the 'setup_build_environment' method.
-        # Copy the source to the installation directory and build OCCA there.
-        install_tree(".", prefix)
-        make("-C", prefix)
+    patch("occa-v1.2.0-cstdint.patch", when="@1.2.0")
+
+    @run_after("install")
+    def check_install(self):
+        version_cmd = join_path(self.prefix.bin, "occa") + " version"
+        val = os.system(version_cmd)
+        if val != 0:
+            raise RuntimeError("Calling `occa version` failed !")
```

Comment:
Can we use the `Executable` class here? Also, shouldn't this be encoded as a test? Wouldn't something ilke `make test` cover this simple case?

Reply:
We don't run OCCA tests as they depend on the GPU computing runtime to be 
available (which sometimes is not available on the build system). Are you referring 
to the `Executable` class [here](https://spack.readthedocs.io/en/latest/spack.util.html#spack.util.executable.Executable)?

Reply:
Yes. You can grep other packages. That class is imported in the recipe via `from spack.package import *`